src/arrayMethodJoin.js

```diff
@@ -4,8 +4,37 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
```

Comment:
No need for this `if`, see Checklist p.5 - Remember that loop with false condition simply will iterate 0 times.

Reply:
5. [CODE STYLE] - don't write loops inside if statement. It's making your code too complicated. Remember that loop with false condition simply will iterate 0 times.
it says not to use a loop inside if statement. Am I misunderstanding something?